src/python/zquantum/core/test_gradients.py

```diff
@@ -0,0 +1,18 @@
+"""Tests for core.gradients module."""
+import numpy as np
+import pytest
+
+from .gradients import finite_differences_gradient
+
+
+def sum_x_squared(parameters: np.ndarray) -> float:
+    return (parameters ** 2).sum()
+
+
+@pytest.mark.parametrize(
+    "parameters", [np.array([1, 2, 3]), np.array([0.5, 0.25, 1, 0.0])]
+)
+def test_finite_differences_gradient_returns_vectors_with_correct_length(parameters):
+    gradient = finite_differences_gradient(sum_x_squared)
+
+    assert len(parameters) == len(gradient(parameters))
```

Comment:
Why are we not testing for the values of the gradients?
And whether `epsilon` actually matters?

Reply:
That's actually a very good question. The reason is: I only ported the tests that were already present in `cost_function_test.py`. But you are right, we can do better. I'll add two more tests checking for the assertions that you mentioned.

Reply:
@mstechly done!